### ISSUE TITLE
BAT tests do not work on ansible installed clusters

### ISSUE THREAD

**Issue by markdryan** [01org/ciao#783]
The reason seems to be that images are not downloaded for all the workloads, for example.

```
$ ciao-cli workload list
Workload 1
	Name: Docker Iperf
	UUID:ab68111c-03a6-11e6-87de-001320fb6e31
	Image UUID: b5e696b8-03a6-11e6-a424-001320fb6e31
	CPUs: 2
	Memory: 128 MB
Workload 2
	Name: Boot Fedora23 from created volume based on image
	UUID:79034317-3beb-447e-987d-4e310a8cf410
	Image UUID: 
	CPUs: 2
	Memory: 128 MB
Workload 3
	Name: Fedora 23 Cloud
	UUID:69e84267-ed01-4738-b15f-b47de06b62e7
	Image UUID: 73a86d7e-93c0-480e-9c41-ab42f69b7799
	CPUs: 2
	Memory: 128 MB
Workload 4
	Name: Clear Cloud
	UUID:e35ed972-c46c-4aad-a1e7-ef103ae079a2
	Image UUID: df3768da-31f5-4ba6-82f0-127a1a705169
	CPUs: 2
	Memory: 128 MB
Workload 5
	Name: Docker Ubuntu latest
	UUID:ca957444-fa46-11e5-94f9-38607786d9ec
	Image UUID: fa7d86d8-fa46-11e5-8493-38607786d9ec
	CPUs: 2
	Memory: 128 MB


$ ciao-cli instance add --workload 79034317-3beb-447e-987d-4e310a8cf410 --instances 1
F1108 11:58:52.897145   31993 instance.go:129] ciao-cli FATAL: HTTP Error [500] for [POST https://ciao-ctl.ld.intel.com:8774/v2.1/d1a1e756c00848809b11e3fcb171d260/servers]: {"error":{"code":500,"name":"Internal Server Error","message":"stat /var/lib/ciao/images/73a86d7e-93c0-480e-9c41-ab42f69b7799: no such file or directory"}}
null
```

**Comment by markdryan** [comment]
@rbradford has pointed out that the issue here is that the images are not copied to the controller node.  This is needed for the boot from volume use case.

**Comment by tpepper** [comment]
Rather than copy them there, I think it's time we go all in on ceph.  Controller just needs to mount the ceph fs of /var/lib/ciao/images today, or later not have it mounted by still have ceph access so it can get at the rbd image by id.

**Comment by albertomurillo** [comment]
I agree with @tpepper its time to stop copying images all over the nodes and just put them once in the cephfs where they are accessible by all nodes.

**Comment by rbradford** [comment]
Please let's not worry about setting up cephfs...remember in our design we've done away with cephfs in favour of rbd only.

My change in #767 requires that controller have access to the images. I'd much rather us have copy the images to the controller node so that we can land that change rather than refactor all our clusters to add cephfs for a short while.

Although it's still a short term change it's much less intrusive that getting cephfs working!

**Comment by tpepper** [comment]
After giving my nodes access to ceph in general, I've just got one line in my /etc/fstab to mount /var/lib/ciao/images.  Really easy incremental.  The ceph connectivity is required either way for cephfs today or rbd in the future.  But it is one thing extra and we could just wait.  The rbd future is getting quite close.

**Comment by kaccardi** [comment]
Possibly a duplicate of #841 - please retest after #841 is closed.

**Comment by albertomurillo** [comment]
With #841 closed the only bat test that are failing are
```
Logs for failed tests
---------------------
Logs for storage_bat.TestCreateBlockDevice
        ceph_test.go:42: Error when running: [qemu-img convert -O rbd /tmp/ciao-random-031297252 rbd:rbd/2b80a6e3-9b6e-4d08-89e9-0729ccf7b5c9:id=ciao]: exit status 1
Logs for storage_bat.TestCopyBlockDevice
        ceph_test.go:65: Error when running: [qemu-img convert -O rbd /tmp/ciao-random-162024179 rbd:rbd/83201699-6e0d-4ac3-8929-179029e75cfd:id=ciao]: exit status 1
```

**Comment by kaccardi** [comment]
the machine that is running the bat tests needs to have qemu-img installed, and have the cephx credentials, so likely you need to ensure that the bat tests are being run from a node inside the ciao cluster, such as the controller node or else make sure that the ceph dependencies are resolved on the host running the tests.

**Comment by albertomurillo** [comment]
Adding cephx and qemu-img to the machine running the tests made the two failing test to pass

```
# test-cases -v ./...
Testing github.com/01org/ciao/_release/bat/base_bat
Testing github.com/01org/ciao/_release/bat/image_bat
Testing github.com/01org/ciao/_release/bat/storage_bat
Package     Test Case                        Time Taken Result
base_bat    TestGetTenants                   0.56s      PASS
base_bat    TestGetClusterStatus             0.27s      PASS
base_bat    TestGetWorkloads                 0.30s      PASS
base_bat    TestGetInstance                  18.64s     PASS
base_bat    TestStartAllWorkloads            24.89s     PASS
base_bat    TestGetCNCIs                     6.04s      PASS
base_bat    TestGetAllInstances              6.34s      PASS
base_bat    TestDeleteAllInstances           6.69s      PASS
image_bat   TestAddShowDelete                2.52s      PASS
image_bat   TestDeleteNonExisting            0.35s      PASS
image_bat   TestImageList                    2.52s      PASS
storage_bat TestCreateBlockDevice            0.58s      PASS
storage_bat TestCopyBlockDevice              1.39s      PASS
storage_bat TestBootFromVolume               3.95s      PASS
storage_bat TestStoppedInstance              10.46s     PASS
storage_bat TestDeleteInUse                  7.55s      PASS
storage_bat TestDetachRootFS                 4.99s      PASS
storage_bat TestAddListDelete                8.89s      PASS
storage_bat TestDeleteInstanceWhileAttaching 4.18s      PASS
storage_bat TestDeleteBeforeDetached         11.09s     PASS
storage_bat TestAttachDetach                 11.90s     PASS
storage_bat TestAttachToRunningContainer     8.52s      PASS
storage_bat TestDoubleAttach                 16.32s     PASS
storage_bat TestAttachToStoppedInstance      13.03s     PASS
```

**Comment by tpepper** [comment]
Should this be closed when there's no docu or automation to insure prerequisites such as cephx and qemu-img stand a better chance of being present?

**Comment by albertomurillo** [comment]
@tpepper This particular issue is resolved but is good that you point that out.
I have opened https://github.com/01org/ciao/pull/935 to update the documentation.
